lms/static/scripts/frontend_apps/components/test/BasicLtiLaunchApp-test.js

```diff
@@ -270,7 +280,7 @@ describe('BasicLtiLaunchApp', () => {
     });
 
     it('displays an error if reporting the submission fails', async () => {
-      const error = new ApiError(400, {});
+      const error = new ApiError(400, { error_message: 'Server error' });
```

Comment:
This test failed after these changes, but I think the test was incorrect. It failed because there is no longer a `retry=false` param that forces a particular dialog that has no retry button. By the looks of it, we don't need that param, so its just simpler to remove it and just assume any error without a message body is an auth error -- this is the assumption I am going on, my my memory is a bit hazy on this.

I did some quick tests and threw faulty params (as a student) and both the 500 and 422 submission errors I saw did  have `error_message` in their response which supports my raional.

Reply:
Looking at https://github.com/hypothesis/lms/blob/e8ac7fc990716185514827a8d0a1d4b476759d94/lms/views/api/error.py#L13, some 4xx errors have an `error_message` property and some only have a `message` property. What happens here if we get an error with a `message` property.

The fact that different JSON 4xx errors use different property names for the error message makes it harder to handle consistently on the frontend. Is that an oversight @seanh?

Reply:
My preference would be to make assumptions from the status code itself if possible.